core/runtime/wavm/impl/intrinsic_functions.cpp

```diff
@@ -0,0 +1,634 @@
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_hashing_blake2_256_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_hashing_keccak_256_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_crypto_ed25519_verify_version_1, WAVM::I32, WAVM::I64, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_misc_runtime_version_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(void, ext_storage_append_version_1, WAVM::I64, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_storage_next_key_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(void, ext_misc_print_num_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_crypto_sr25519_verify_version_2, WAVM::I32, WAVM::I64, WAVM::I32);
+    GENERATE_STUB_INTRINSIC(void, ext_offchain_local_storage_set_version_1, WAVM::I32, WAVM::I64, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_storage_root_version_1);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_storage_changes_root_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_trie_blake2_256_ordered_root_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_crypto_ed25519_generate_version_1, WAVM::I32, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_crypto_secp256k1_ecdsa_recover_version_1, WAVM::I32, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_crypto_secp256k1_ecdsa_recover_compressed_version_1, WAVM::I32, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_crypto_sr25519_generate_version_1, WAVM::I32, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_crypto_sr25519_public_keys_version_1, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_crypto_sr25519_sign_version_1, WAVM::I32, WAVM::I32, WAVM::I64);
+    GENERATE_STUB_INTRINSIC(WAVM::I64, ext_offchain_network_state_version_1);
+    GENERATE_STUB_INTRINSIC(WAVM::I64, ext_offchain_submit_transaction_version_1, WAVM::I64);
+    GENERATE_HOST_INTRINSIC(WAVM::I64, ext_storage_read_version_1, WAVM::I64, WAVM::I64, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(WAVM::I32, ext_allocator_malloc_version_1, WAVM::I32);
+    GENERATE_HOST_INTRINSIC(void, ext_allocator_free_version_1, WAVM::I32);
+    // clang-format on
+  }
+
+#undef WAVM_DEFINE_INTRINSIC_FUNCTION
+#define WAVM_DEFINE_INTRINSIC_FUNCTION(module, nameString, Result, cName, ...) \
+  Result cName(WAVM::Runtime::ContextRuntimeData *contextRuntimeData,          \
+               ##__VA_ARGS__);                                                 \
+  WAVM::Intrinsics::Function cName##Intrinsic(                                 \
+      getIntrinsicModule_##module(),                                           \
+      nameString,                                                              \
+      (void *)&cName,                                                          \
+      WAVM::Intrinsics::inferIntrinsicFunctionType(&cName));                   \
+  Result cName(WAVM::Runtime::ContextRuntimeData *contextRuntimeData,          \
```

Comment:
I don't see the use of ContextRuntimeData inside function. I suppose it is obligatory by interface?

Reply:
Correct. Will leave a comment.